ra/registration-authority.go

```diff
@@ -602,6 +608,11 @@ func (ra *RegistrationAuthorityImpl) UpdateRegistration(base core.Registration,
 
 // UpdateAuthorization updates an authorization with new values.
 func (ra *RegistrationAuthorityImpl) UpdateAuthorization(base core.Authorization, challengeIndex int, response core.Challenge) (authz core.Authorization, err error) {
+	// Refuse to update expired authorizations
+	if base.Expires == nil || base.Expires.Before(time.Now()) {
+		err = core.MalformedRequestError("Expired authorization")
```

Comment:
I don't think it really makes sense to push that logic into the RA.  The "gone-ness" is a property of the authz/challenge as a web resource, so the WFE should handle it.  I'll just have the RA return a plain error that will turn into a 500, since this shouldn't happen.